### DISCUSSION TITLE
Is Capacitor google maps compatible with JDK 20 ?

### DISCUSSION THREAD

**Discussion by Martial203** [ionic-team/capacitor#6994] [Q&A]
I tried using it on android, but my app doesn't compile. I use JDK 20 and Gradle 8.0.2. All works perfectly in browser, but the app doesn't build for android, with the following message :

':capacitor-google-maps:compileDebugKotlin'.
        > Failed to calculate the value of property 'jvmTarget'.        
        > Unknown Kotlin JVM target: 20

Please, need for help.

I'm looking for a solution since 02 days. I tried many solution proposals, but nothing worked for me yet

**Comment by jcesarmobile** [comment]
Android development requires java 17 at the moment.
java 20 will be supported on gradle 8.3.

**Reply by Martial203** [reply]
I downgraded my jdk to jdk 17, the app build, but crash after splash screen. Now, I'm trying to upgrade my gradle to gradle 8.3, to try it again with jdk 20

**Comment by Martial203** [comment]
I upgraded gradle to gradle 8.3, but I get the same error.

So, I'll go back to jdk 17 and try to fix the crash encountered by the app after the splash screen.

Please any solution or indication to help me fix it ? 🙏

**Reply by jcesarmobile** [reply]
hard to tell you if you don't provide the error, check logcat as it should be printed there

**Reply by Martial203** [reply]
Thank you very much, it's okay. I used the logcat and I noticed that I did not provide the api key of a plugin in the right place, in my Android Manifest.

Now, all is fine, thank you very much.

**Comment by renasance** [comment]
Hi Im experiencing the same problem. Funny enough I was able to run it externally on my phone. Then recently, I havent kept up with my updates, but I have the same error. Today, Gradle 8.3 can support Java 20. It runs fine in browser, but running it externally yields me the same error:

 ':capacitor-google-maps:compileDebugKotlin'
> Failed to calculate the value of property 'jvmTarget'.
> Unknown Kotlin JVM target: 20

Anyone else experiencing this? Or have a solution?
